### ISSUE TITLE
Encouraging Gaussians to be 2D

### ISSUE THREAD

**Issue by mira-s** [maturk/dn-splatter#111]
Hello, thanks for the solid papers and useful repo!
The concept of encouraging Gaussians to be 2D seems very useful, however, I cannot find where `--pipeline.model.two_d_gaussians` gets used. I only see it defined in `DNSplatterModelConfig`, but not used, and it's not a parameter in the base `SplatfactoModelConfig`. Is some part of the implementation missing?

**Comment by maturk** [comment]
Hi, the 2D Gaussian loss is applied [here](https://github.com/maturk/dn-splatter/blob/249d52c4bb14b7bf6dd18d7d66099a36eac2ee78/dn_splatter/regularization_strategy.py#L195) for the dn-splatter model. It encourages the minimum scaling axis to approach zero. Let me know if you have any other doubts. 

Best,
maturk

**Comment by mira-s** [comment]
I see, so it's independent from the `--pipeline.model.two_d_gaussians` parameter and is always part of the regularization loss, and also different from `splatfacto`'s `scale_reg`. Thanks a lot for the pointers!

**Comment by maturk** [comment]
Yea looks like a bug, I refactored the code-base some time ago and looks like I forgot to pass the two_d_gaussians parameter to the regularization strategy. It is on by default. 

And yes, this is different from splatfacto's scale_reg. There the regularization is between the ratio of scales, trying to make sure that there are no long/skinny gaussians. But the scale loss in dn-splatter simply minimizes the scaling axis corresponding to the normal direction.

**Comment by mira-s** [comment]
Makes sense, thanks for the extra details!

**Comment by maturk** [comment]
Np

**Comment by Irving87** [comment]
> Yea looks like a bug, I refactored the code-base some time ago and looks like I forgot to pass the two_d_gaussians parameter to the regularization strategy. It is on by default.
> 
> And yes, this is different from splatfacto's scale_reg. There the regularization is between the ratio of scales, trying to make sure that there are no long/skinny gaussians. But the scale loss in dn-splatter simply minimizes the scaling axis corresponding to the normal direction.

@maturk Hello, may I ask how "the scale loss in dn-splatter simply minimizes the scaling axis corresponding to the normal direction"? I found the code  `get_loss_dict`  function in dn_model.py:
`additional_data = {
            "scales": self.scales,
            "gt_img": gt_img,
        }`
```python
if self.config.regularization_strategy == dn-splatter:
            regularization_strategy_loss = self.regularization_strategy(
                pred_depth=depth_out,
                gt_depth=depth_gt,
                pred_normal=pred_normal,
                gt_normal=gt_normal,
                **additional_data,
            )
```
and this will call the function:
```python
def get_loss(self, pred_depth, gt_depth, pred_normal, gt_normal, **kwargs):
        depth_loss, normal_loss = 0.0, 0.0
        if self.depth_loss is not None:
            depth_loss = self.get_depth_loss(pred_depth, gt_depth, **kwargs)
        if self.normal_loss is not None:
            normal_loss = self.get_normal_loss(pred_normal, gt_normal, **kwargs)
        scales = kwargs["scales"]
        scale_loss = self.get_scale_loss(scales=scales)
        return depth_loss + normal_loss + scale_loss
```
But in the `get_scale_loss` function, it seems to only calculate the minimum value in the scale:
```python
def get_scale_loss(self, scales):
        """Scale loss"""
        # loss to minimise gaussian scale corresponding to normal direction
        scale_loss = torch.min(torch.exp(scales), dim=1, keepdim=True)[0].mean()
        return scale_loss
```

**So my question is, how the gaussian scale minimised corresponding to normal direction in dn_stlatter?** I cannot find the code how the scale minimised according to normal direction, could you tell my where these codes are located? Thanks a lot!
